setup/steps/09_deploy_via_modelservice.py

```diff
@@ -0,0 +1,693 @@
+    extract_environment()
+    
+    # Extract flags
+    dry_run = ev.get("control_dry_run", "false") == "true"
+    verbose = ev.get("control_verbose", "false") == "true"
+    
+    # Deploy models
+    model_list = ev.get("deploy_model_list", "").replace(",", " ").split()
+    model_number = 0
+    
+    for model in model_list:
+        if not model.strip():
+            continue
+            
+        # Set current model environment variables
+        current_model = model_attribute(model, "model")
+        current_model_id = model_attribute(model, "modelid")
+        current_model_id_label = model_attribute(model, "modelid_label")
+        
+        os.environ["LLMDBENCH_DEPLOY_CURRENT_MODEL"] = current_model
+        os.environ["LLMDBENCH_DEPLOY_CURRENT_MODEL_ID"] = current_model_id
+        os.environ["LLMDBENCH_DEPLOY_CURRENT_MODEL_ID_LABEL"] = current_model_id_label
+        
+        # Update ev dictionary with new model info
+        ev["deploy_current_model"] = current_model
+        ev["deploy_current_model_id"] = current_model_id
+        ev["deploy_current_model_id_label"] = current_model_id_label
+        
+        # Determine model mounting
+        mount_model_volume = False
+        if (ev.get("vllm_modelservice_uri_protocol") == "pvc" or 
+            ev.get("control_environment_type_standalone_active", "0") == "1"):
+            pvc_name = ev.get("vllm_common_pvc_name", "")
+            os.environ["LLMDBENCH_VLLM_MODELSERVICE_URI"] = f"pvc://{pvc_name}/models/{current_model}"
+            mount_model_volume = True
+        else:
+            os.environ["LLMDBENCH_VLLM_MODELSERVICE_URI"] = f"hf://{current_model}"
+            mount_model_volume = True
+        
+        # Check for mount override
+        mount_override = ev.get("vllm_modelservice_mount_model_volume_override")
+        if mount_override:
+            mount_model_volume = mount_override == "true"
+        
+        # Update ev with URI
+        ev["vllm_modelservice_uri"] = os.environ["LLMDBENCH_VLLM_MODELSERVICE_URI"]
+        
+        # Create directory structure (Do not use "llmdbench_execute_cmd" for these commands)
+        model_num = f"{model_number:02d}"
+        release = ev.get("vllm_modelservice_release", "")
+        work_dir = Path(ev.get("control_work_dir", ""))
+        helm_dir = work_dir / "setup" / "helm" / release / model_num
+        
+        # Always create directory structure (even in dry-run)
+        helm_dir.mkdir(parents=True, exist_ok=True)
+        
+        # Generate ms-rules.yaml content first
+        rules_content = generate_ms_rules_yaml(ev, model_number)
+        rules_file = helm_dir / "ms-rules.yaml"
+        
+        # Write empty file first, then write content
+        rules_file.write_text("")
+        if len([m for m in model_list if m.strip()]) == 1:
+            rules_file.write_text(rules_content)
+        
+        # Set up configuration preparation
+        add_config_prep()
```

Comment:
Given the proposed changes earlier (comment: `# Extra configurations`), this is no longer needed. Nor is the method.